apps/opik-backend/src/main/java/com/comet/opik/domain/TraceService.java

```diff
@@ -90,35 +92,57 @@ public Mono<UUID> create(@NonNull Trace trace) {
 
         String projectName = WorkspaceUtils.getProjectName(trace.projectName());
         UUID id = trace.id() == null ? idGenerator.generateId() : trace.id();
+        var finalTrace = trace.toBuilder().lastUpdatedAt(Instant.now()).build();
 
-        return IdGenerator
+        return AsyncUtils.makeMonoContextAware((userName, workspaceId) -> IdGenerator
                 .validateVersionAsync(id, TRACE_KEY)
                 .then(Mono.defer(() -> getOrCreateProject(projectName)))
                 .flatMap(project -> lockService.executeWithLock(
                         new LockService.Lock(id, TRACE_KEY),
-                        Mono.defer(() -> insertTrace(trace, project, id))));
+                        insertTrace(finalTrace, project, id)
+                                .then(Mono.fromRunnable(() -> projectService.recordLastUpdatedTrace(
+                                        workspaceId, List.of(new ProjectIdLastUpdated(
+                                                project.id(), finalTrace.lastUpdatedAt())))))
```

Comment:
This will slow down the trace ingestion, we can do this using the event bus. Please see the experiment creation